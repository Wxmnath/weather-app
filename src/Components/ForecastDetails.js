import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

function ForecastDetails(props) {
  const { date, temperature, humidity, wind } = props;

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>

      <div className="forecast-summary__max-temperature">
        {temperature.max}
        &deg;C
      </div>

      <div className="forecast-summary__min-temperature">
        {temperature.min}
        &deg;C
      </div>

      <div className="forecast-details__humidity">{humidity} </div>

      <div className="forecast-summary__wind">{wind.speed}</div>

      <div className="forecast-summary__direction">{wind.direction}</div>
    </div>
  );
}

ForecastDetails.propTypes = {
  date: PropTypes.number.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
  humidity: PropTypes.number.isRequired,
  wind: PropTypes.shape({
    speed: PropTypes.number,
    direction: PropTypes.string,
  }).isRequired,
};

export default ForecastDetails;
