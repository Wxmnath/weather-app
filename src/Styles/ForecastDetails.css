.forecast-details {
  text-align: center;
  margin-top: 1rem;
  border-radius: 20px 5px 20px;
  border: solid blue;
  padding: 25px 0;
  color: blue;
  line-height: 1.2;
  background-color: lightgray;
}

.forecast-details__date {
  font-size: 50px;
  margin-top: 0.5rem;
  margin-bottom: 1rem;
}
