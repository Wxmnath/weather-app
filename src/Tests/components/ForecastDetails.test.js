import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../Components/ForecastDetails";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 80,
    wind: {
      speed: 13,
      direction: "NW",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        temperature={validProps.temperature}
        humidity={validProps.humidity}
        wind={validProps.wind}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
