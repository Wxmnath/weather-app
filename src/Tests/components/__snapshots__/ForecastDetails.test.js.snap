// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-summary__max-temperature"
    >
      22°C
    </div>
    <div
      class="forecast-summary__min-temperature"
    >
      12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      80 
    </div>
    <div
      class="forecast-summary__wind"
    >
      13
    </div>
    <div
      class="forecast-summary__direction"
    >
      NW
    </div>
  </div>
</DocumentFragment>
`;
